### PR TITLE
Region is configurable during IAM Wizard and Cloudformation Stackset flow

### PR DESCRIPTION
**Pull request by Will-NOQ**
## What changed?
* Updated the wizard and cloud formation functions to make the AWS region IAMbic will use configurable.

## Rationale
* Organizations with an SCP that prevents usage in us-east-1 needs a way to define their preferred region

## How was it tested?
If it was manually verified, list the instructions for your reviewers to follow.
- [x] Unit Tests
- [x] Functional Tests
- Went through setup using an org with import
- Went through setup using an account with import

### CODE REVIEW

File: iambic/plugins/v0_1_0/aws/cloud_formation/templates/IambicSpokeRole.yml [hunk 1]
```diff
@@ -30,6 +30,13 @@ Resources:
           PolicyDocument:
             Version: '2012-10-17'
             Statement:
+              - Effect: Allow
```
Comment (smoy) [inline iambic/plugins/v0_1_0/aws/cloud_formation/templates/IambicSpokeRole.yml:33]: this is already covered in Sid `SqsWriteAccessForResourceMonitoring `

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,26 @@
-## 0.4.1 (Target Date May 1st, 2023)
+
+## 0.5.1 (Target Date May 2nd, 2023)
```
Comment (castrapel) [inline CHANGELOG.md:2]: Should this be 0.4.2 or is it worthy of a minor bump?

Comment (Will-NOQ) [reply CHANGELOG.md:2]: tbh I was just preserving the version @smoy was using

File: iambic/plugins/v0_1_0/aws/cloud_formation/templates/IambicSpokeRole.yml (hunk 1 above)
Comment (Will-NOQ) [reply iambic/plugins/v0_1_0/aws/cloud_formation/templates/IambicSpokeRole.yml]: Good catch, fixed!

File: iambic/plugins/v0_1_0/azure_ad/models.py [hunk 3]
```diff
@@ -140,6 +140,8 @@ def dict(
             exclude = required_exclude
         elif isinstance(exclude, set):
             exclude.update(required_exclude)
+        # elif isinstance(exclude, dict):
```
Comment (castrapel) [inline iambic/plugins/v0_1_0/azure_ad/models.py:143]: Safe to assume this isn't needed anymore / wasn't removed temporarily for debugging purposes?

Comment (Will-NOQ) [reply iambic/plugins/v0_1_0/azure_ad/models.py]: Oops, yeah just removed it

**Review by castrapel — APPROVED** [review]
LGTM

**Pull request merged**